homeassistant/components/zeroconf/__init__.py

```diff
@@ -45,3 +81,27 @@
     hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_zeroconf)
 
     return True
+
+
+def info_from_service(service):
+    """Return most important info from mDNS entries."""
+    properties = {}
+
+    for key, value in service.properties.items():
+        if isinstance(value, bytes):
+            value = value.decode('utf-8')
+        properties[key.decode('utf-8')] = value
+
+    address = service.address or service.address6
+
+    info = {
+        ATTR_HOST: str(ipaddress.ip_address(address)),
+        ATTR_PORT: service.port,
+        ATTR_HOSTNAME: service.server,
+        ATTR_PROPERTIES: properties,
+    }
+
+    if "mac" in properties:
+        info[ATTR_MAC_ADDRESS] = properties["mac"]
```

Comment:
I don't think this is needed since we know it's inside the properties. Only things that are not in properties we should add to info. 

Great that MAC is there btw, we can store it in the config entry for unique ID :D

Reply:
We should try to store as much info on info as possible. I wonder if we should just consider to pass the info object? Although the properties being bytes is annoying.

Reply:
Removing separate Mac

That is all information already; https://github.com/frawau/aiozeroconf/blob/9c473cc2bc76622bdfa5063422e80699582451c4/aiozeroconf/aiozeroconf.py#L1266